### ISSUE TITLE
A drive with this name already exists

### ISSUE THREAD

**Issue by Badgerati** [Badgerati/Pode#1029]
I was sure I'd fixed this one, but I just got the `A drive with name already exists` error again from testing Pode.Web - on linux.

```plain
pode-web-example  | New-PSDrive: /usr/local/share/powershell/Modules/Pode/Private/Helpers.ps1:853
pode-web-example  | Line |
pode-web-example  |  853 |  …   $drive = (New-PSDrive -Name $Name -PSProvider FileSystem -Root $Pat …
pode-web-example  |      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pode-web-example  |      | A drive with the name
pode-web-example  |      | 'PodeDir45ee17d2-3c82-4958-97f6-7948622147d6' already exists.
pode-web-example  |
pode-web-example  | at New-PodePSDrive, /usr/local/share/powershell/Modules/Pode/Private/Helpers.ps1: line 853
pode-web-example  | at Add-PodePSDrives, /usr/local/share/powershell/Modules/Pode/Private/Helpers.ps1: line 905
pode-web-example  | at Open-PodeRunspace, /usr/local/share/powershell/Modules/Pode/Private/Helpers.ps1: line 604
pode-web-example  | at <ScriptBlock>, <No file>: line 1
```

**Comment by jbaechtelMT** [comment]
I had this save error and corrected it by commenting out (add # at the beginning of the line) line 853 in Helpers.ps1.  Then add this line just below it:

if (!(Get-PSDrive -Name $Name -ErrorAction SilentlyContinue)) {

This will fix you problem.  Enjoy!

**Comment by jbaechtelMT** [comment]
@Badgerati - The true error is in the **Test-PodePSDrive** function.  My response (above) is just a work around for now.  THX.  Keep up the good work!

**Comment by ittchmh** [comment]
I saw this error only if relative path to folder was used, if absolute path is used - no issues

**Comment by ili101** [comment]
Badgerati I think adding `Resolve-Path` could be a solution
```powershell
[DBG]: [Process:47918]: [Runspace40]: PS /PsF/Scripts/Tool>> $drive.Root
/PsF/Scripts/Tool/Logs
[DBG]: [Process:47918]: [Runspace40]: PS /PsF/Scripts/Tool>> $Path                                
/PsF/Scripts/Tool/./Logs
[DBG]: [Process:47918]: [Runspace40]: PS /PsF/Scripts/Tool>> $drive.Root -ieq $Path               
False
[DBG]: [Process:47918]: [Runspace40]: PS /PsF/Scripts/Tool>> $drive.Root -ieq (Resolve-Path $Path)
True
```

**Comment by Badgerati** [comment]
Aah! Yes, you're both right!

Adding a `Resolve-Path` into `New-PodePSDrive` should resolve this then 😄

**Comment by Badgerati** [comment]
The above commit should now resolve this; I've tested locally where I could always get the error to trip, but no linger see it with the changes.

I've also altered references of `Resolve-Path` to `[System.IO.Path]::GetFullPath(...)` as it's substantially faster.

The main change is to add `$Path = [System.IO.Path]::GetFullPath($Path)` just before the `if (!(Test-PodePSDrive -Name $Name -Path $Path)) {` line in `New-PodePSDrive`. If anyone's able to test this themselves let me know 😄
